src/utilities/extendSchema.js

```diff
@@ -216,9 +227,9 @@ export function extendSchema(
     subscription: extendMaybeNamedType(schema.getSubscriptionType()),
   };
 
-  // Then, incorporate all schema extensions.
-  for (const schemaExtension of schemaExtensions) {
-    if (schemaExtension.operationTypes) {
+  // Then, incorporate schema definition and all schema extensions.
+  for (const schemaExtension of [schemaDef, ...schemaExtensions]) {
```

Comment:
oh wait just saw this: can we do instead:
```
if (schemaDef) {
  schemaExtensions.push(schemaDef);
}
for (const schemaExtension of schemaExtensions) {
```

Reply:
Main worry here: it seems silly to do null checking inside the for loop instead of making sure what we have in the array we're looping over is non-null.

Reply:
@mjmahone You are right, the current code is confusing 😞 
Sadly, `schemaExtensions` used bellow explicitly for extension nodes.
Moreover `schemaDef` doesn't need `schemaDef.operationTypes` because it always initialized. 
So I decided to just duplicate code since #1435 will reduce this code and we can simplify things in subsequent refactorings.